### ISSUE TITLE
Build fails with checkout v2

### ISSUE THREAD

**Issue by gitfool** [actions/checkout#113]
See [gitfool/Cake.Dungeon](https://github.com/gitfool/Cake.Dungeon) workflow builds for commit [7963aee](https://github.com/gitfool/Cake.Dungeon/commit/7963aee4105bfcaad15c3a994597a5ed031d43da):
- [Linux](https://github.com/gitfool/Cake.Dungeon/runs/348899476)
- [Mac](https://github.com/gitfool/Cake.Dungeon/runs/348899466)
- [Windows](https://github.com/gitfool/Cake.Dungeon/runs/348899478)

Failing in with an obscure error:
```
Error: One or more errors occurred. (Expecting state 'Element'.. Encountered 'Text'  with name '', namespace ''.)
	Expecting state 'Element'.. Encountered 'Text'  with name '', namespace ''.
##[error]Process completed with exit code 1.
```

If I revert to checkout v1 it builds successfully again.

**Comment by ericsciple** [comment]
Need more info.

Is something different between the layout when using v2 vs v1, that is causing your build step to fail?

Otherwise does your build require additional git history? V2 was optimized for the mainline and only fetches a single commit rather than all history.

Note, the error sounds related to reading an xml file.

**Comment by gitfool** [comment]
@ericsciple sorry about that; more info revealed after enabling Cake diagnostic verbosity. I'm using [GitVersion](https://github.com/GitTools/GitVersion) and it clearly doesn't like "shallow" checkout v2:
```
ERROR [12/16/19 2:20:28:11] An unexpected error occurred:
2019-12-16T02:20:28.1401577Z GitVersion.Exceptions.GitVersionException:
Can't find commit d8aabf404b69d615cba1c9f13e2ee975e6560c6d.
Please ensure that the repository is an unshallow clone with `git fetch --unshallow`.
```
Full error in log [here](https://github.com/gitfool/Cake.Dungeon/runs/349867050#step:6:226). I'm not sure if GitVersion can do anything about this, so is there something that could be made configurable to help in checkout v2?

**Comment by ericsciple** [comment]
@gitfool oh gotcha. V2 was optimized to fetch a single commit. If you need more history for the branch being fetched, then increase the `fetch-depth` (defaults to 1, set to 0 for all history)

Otherwise if you need all history, i think `run: git fetch --unshallow` will fetch all branches and tags (all commits)

I'll update the docs to include an example to fetch all history for all branches and tags.

**Comment by ericsciple** [comment]
Updated the docs: https://github.com/actions/checkout/pull/115

**Comment by gitfool** [comment]
@ericsciple note that `git fetch --unshallow` is not sufficient for GitVersion as tags are _not_ fetched unless `--tags` is also specified, so the documentation is misleading.

As I just mentioned [here](https://github.com/actions/checkout/issues/100#issuecomment-569443501), it would be much nicer to have a `fetch-tags` option in checkout v2. That way it uses the recommended defaults but can be tweaked to get v1 behaviour without losing other v2 benefits and without needing a separate `run` command.

**Comment by AArnott** [comment]
@ericsciple Can you please add a parameter to `actions/checkout@v2` that instructs the original clone to fetch the full history of HEAD at least? Having a separate fetch step to get more commits seems overly burdensome for folks who wanted the `@v1` behavior.

**Comment by ericsciple** [comment]
@AArnott the [`fetch-depth` parameter](https://github.com/actions/checkout#usage) can be used

**Comment by AArnott** [comment]
Oh sheesh. I don't know how I missed that in the README. Thanks, @ericsciple.

**Comment by gitfool** [comment]
@ericsciple the `fetch-depth` parameter is not sufficient in my case since it always specifies [--no-tags](https://github.com/actions/checkout/blob/master/src/git-command-manager.ts#L146). Could you please make this configurable too?

**Comment by ericsciple** [comment]
i have some changes in mind, collecting some feedback from my immediate team and will publish a proposal soon

**Comment by gitfool** [comment]
@ericsciple gentle nudge; have you made any progress on the above? 🙏

**Comment by ericsciple** [comment]
@gitfool soliciting feedback on [this PR](https://github.com/actions/checkout/pull/155). sorry for the delay, other things came up.

**Comment by pharos-sfausett** [comment]
@ericsciple no worries. My use case is that I currently need to follow checkout with:
```yaml
- name: Fetch unshallow
  run: git fetch --prune --tags --unshallow
```
in order to get full history as needed by [GitVersion](https://github.com/GitTools/GitVersion). Needless to say, I'd love to get rid of this extra step.

**Comment by gitfool** [comment]
@ericsciple no worries. My use case is that I currently need to follow checkout with:
```yaml
- name: Fetch unshallow
  run: git fetch --prune --tags --unshallow
```
in order to get full history as needed by [GitVersion](https://github.com/GitTools/GitVersion). Needless to say, I'd love to get rid of this extra step.

**Comment by bastimeyer** [comment]
Using v2 is very clunky right now when your build steps require the output of `git describe --tags` for example. I would appreciate an option that would let me fetch tags in addition to the fetch depth, without having to define a separate run step. Thank you.

**Comment by ericsciple** [comment]
@bastimeyer i agree. You should +1 this proposal https://github.com/actions/checkout/pull/155 :)

**Comment by GiorgioBertolotti** [comment]
> @ericsciple no worries. My use case is that I currently need to follow checkout with:
> 
> 
> - name: Fetch unshallow
>   run: git fetch --prune --tags --unshallow
> in order to get full history as needed by [GitVersion](https://github.com/GitTools/GitVersion). Needless to say, I'd love to get rid of this extra step.

That fixes my problem, thanks @gitfool

**Comment by gitfool** [comment]
@GiorgioBertolotti that hack is no longer needed. See [gitfool/Cake.Dungeon@9713d4f](https://github.com/gitfool/Cake.Dungeon/commit/9713d4fdb64327a4f3ae533c4943d94e0d3c6115#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721).
